kuegi_bot/backtest_engine.py

```diff
@@ -198,17 +210,32 @@ def handle_open_orders(self, intrabarToCheck: Bar) -> bool:
                             should_execute= True
                 else:  # means order.limit_price and (order.stop_price is None or order.stop_triggered):
                     # check for limit execution
-                    if (order.amount > 0 and order.limit_price > intrabarToCheck.low) or (
-                            order.amount < 0 and order.limit_price < intrabarToCheck.high):
+                    ref = intrabarToCheck.low if order.amount > 0 else intrabarToCheck.high
+                    if onlyOnClose:
+                        ref= intrabarToCheck.close
+                    if (order.amount > 0 and order.limit_price > ref) or (
+                            order.amount < 0 and order.limit_price < ref):
                         should_execute= True
 
                 if should_execute:
                     self.handle_order_execution(order, intrabarToCheck)
                     self.bot.on_tick(self.current_bars, self.account)
                     another_round= True
-                    something_changed= True
+                    didSomething= True
                     break
+        return didSomething
+
+    def handle_subbar(self, intrabarToCheck: Bar):
+        # first the ones that are there at the beginning
+        orders= list(self.account.open_orders)
+        didSomething1= self.check_executions(intrabarToCheck,False)
+        self.current_bars[0].add_subbar(intrabarToCheck)  # so bot knows about the current intrabar
+
+        didSomething2= self.check_executions(intrabarToCheck,True)
```

Comment:
didSomething1 and 2? is there no more meaningful meaning for those?

anyways would be did_something_1 and did_something_2